website/source/docs/guides/acl-migrate-tokens.html.md

```diff
@@ -0,0 +1,453 @@
+have.
+
+#### Strategy 1: Simple Policy Mapping
+
+The simplest and most automatic strategy is to create one new policy for every
+existing token. This is easy to automate, but may result in a lot of policies
+with exactly the same rules and with non-human-readable names which will make
+managing policies harder. This approach can be accomplished using the [`consul
+acl policy create`](/docs/commands/acl/acl-policy.html#create) command with
+`-from-token` option.
+
+| Pros | Cons |
+| ---- | ---- |
+| &#9989; Simple           | &#10060; May leave many duplicated policies |
+| &#9989; Easy to automate | &#10060; Policy names not human-readable |
+
+A detailed example of using this approach is [given
+below](#simple-policy-mapping).
+
+#### Strategy 2: Combining Policies
+
+This strategy takes a more manual approach to create a more manageable set of
+policies. There are a spectrum of options for how to do this which tradeoff
+increasing human involvement for increasing clarity and re-usability of the
+resulting policies.
+
+For example you could use hashes of the policy rules to de-duplicate identical
+token policies automatically, however naming them something meaningful for
+humans would likely still need manual intervention.
+
+Toward the other end of the spectrum it might be beneficial for security to
+translate prefix matches into exact matches. This however requires the operator
+knowing that clients using the token really doesn't rely on the prefix matching
+semantics of the old ACL system.
+
+To assist with this approach, there is a CLI tool and corresponding API that can
+translate a legacy ACL token's rules into a new ACL policy that is exactly
+equivalent. See [`consul acl
+translate-rules`](/docs/commands/acl/acl-translate-rules.html).
+
+| Pros | Cons |
+| ---- | ---- |
+| &#9989; Clearer, more manageable policies | &#10060; Requires more manual effort |
+| &#9989; Policies can be re-used by new ACL tokens | &#10060; May take longer for large or complex existing policy sets |
+
+A detailed example of using this approach is  [given below](#combining-policies).
+
+### Updating Existing Tokens
+
+Once you have created one or more policies that adequately express the rules
+needed for a legacy token, you can update the token via the CLI or API to use
+those policies.
+
+After updating, the token is no longer considered "legacy" and will have all the
+properties of a new token, however it keeps it's `SecretID` (the secret part of
+the token used in API calls) so clients already using that token will continue
+to work. It is assumed that the policies you attach continue to grant the
+necessary access for existing clients; this is up to the operator to ensure.
+
+#### Update via API
+
+Use the [`PUT /v1/acl/token/:AccessorID`](/api/acl/tokens.html#update-a-token)
+endpoint. Specifically, ensure that the `Rules` field is omitted or empty. Empty
+`Rules` indicates that this is now treated as a new token.
```

Comment:
Does the new `token` endpoint even accept `Rules`? I thought it didn't? Maybe I'm wrong

Reply:
Kinda! It parses them out, but it errors if it's non-empty. This is more in reference to our `api` package which uses the same request object which still has a `Rules` field so you have to take care to unset that if you are using the same object you got back from the List API to start with...